integration-tests/vertx-http-compressors/app/src/test/java/io/quarkus/compressors/it/Testflow.java

```diff
@@ -0,0 +1,112 @@
+package io.quarkus.compressors.it;
+
+import static io.netty.handler.codec.compression.ZlibCodecFactory.newZlibDecoder;
+import static io.netty.handler.codec.http.HttpResponseStatus.OK;
+import static io.quarkus.arc.ComponentsProvider.LOG;
+import static io.quarkus.compressors.it.CompressedResource.TEXT;
+import static java.lang.Integer.parseInt;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.embedded.EmbeddedChannel;
+import io.netty.handler.codec.compression.BrotliDecoder;
+import io.netty.handler.codec.compression.ZlibWrapper;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.WebClientOptions;
+
+public class Testflow {
+
+    /**
+     * This test logic is shared by both "all" module and "some" module.
+     * See their RESTEndpointsTest classes.
+     *
+     * @param endpoint
+     * @param acceptEncoding
+     * @param contentEncoding
+     * @param contentLength
+     */
+    public static void runTest(String endpoint, String acceptEncoding, String contentEncoding, String contentLength) {
+        LOG.infof("Endpoint %s; Accept-Encoding: %s; Content-Encoding: %s; Content-Length: %s",
+                endpoint, acceptEncoding, contentEncoding, contentLength);
+        // RestAssured
+        // -----------
+        // Why not use RestAssured? Because it doesn't let you configure Accept-Encoding easily
```

Comment:
Should an issue be raised with RestAssured to improve this?

Reply:
@galderz Being a test framework, I think it could do better at letting you tweak the headers manually. The topic is not new to the RestAssured GitHub though. I will make a TODO to follow up.